app/custom-oauth/server/oauth_helpers.js

```diff
@@ -40,3 +43,31 @@ export function updateRolesFromSSO(user, identity, roleClaimName) {
 		});
 	}
 }
+
+export function mapSSOGroupsToChannels(user, identity, groupClaimName, channelsMap, channelsAdmin) {
+	if (user && identity && groupClaimName) {
+		const groupsFromSSO = identity[groupClaimName] || [];
+
+		for (const ssoGroup in channelsMap) {
+			if (typeof ssoGroup === 'string') {
+				let channels = channelsMap[ssoGroup];
+				if (!Array.isArray(channels)) {
+					channels = [channels];
+				}
+				for (const channel of channels) {
+					let room = Rooms.findOneByNonValidatedName(channel);
+					if (!room) {
+						room = createRoom('c', channel, channelsAdmin, [], false);
```

Comment:
Is this save to be doing here?  When does this run?  Isn't this running every time an oauth request comes in?

Reply:
it should run on every oauth request, to in case apply updates